### PR TITLE
Support offset pitch for `Safe mode` and `NSM` events and improve `Safe mode`

### PR DESCRIPTION
**Pull request by taldcroft**
## Description

- Add support for a `pitch` parameter in the `NSM` and `Safe mode` events.
- Improve the fidelity of the `Safe mode` event. 
  - Previously the command set for safe mode started with two safe mode commands and then included all commands for a NSM transition, including the `AONSMSAF` command (in order to tell states to do the pitch maneuver).
  - Now kadi states understands natively the safe mode command `ACPCSFSU` and initiates a maneuver accordingly.

### Unrelated changes

- Fix documentation so that making a configuration file works.
- ~~Sort parameters in `pprint_like_backstop()` so that comparing two backstops has minimal diffs.~~

### Unit tests
There is a unit test for this functionality in #340. It was just easier to roll it into that PR.

- [x] Mac
```
(ska3) ➜  kadi git:(matlab-support-nsm-ssm-offsets) git rev-parse HEAD
587015c61df212f2073c8d5e827181265b99f447
(ska3) ➜  kadi git:(matlab-support-nsm-ssm-offsets) pytest
============================================ test session starts ============================================
platform darwin -- Python 3.11.8, pytest-7.4.4, pluggy-1.4.0
rootdir: /Users/aldcroft/git
configfile: pytest.ini
plugins: timeout-2.2.0, anyio-4.3.0
collected 181 items                                                                                         

kadi/commands/tests/test_commands.py ................................................................ [ 35%]
.................                                                                                     [ 44%]
kadi/commands/tests/test_states.py .......................x.........................                  [ 71%]
kadi/commands/tests/test_validate.py ...................                                              [ 82%]
kadi/tests/test_events.py ..........                                                                  [ 87%]
kadi/tests/test_occweb.py ......................                                                      [100%]

================================ 180 passed, 1 xfailed in 113.45s (0:01:53) =================================
```

Independent check of unit tests by Jean
- [x] Linux
```
(ska3-masters) jeanconn-fido> pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.11.8, pytest-7.4.4, pluggy-1.4.0
rootdir: /proj/sot/ska/jeanproj/git
configfile: pytest.ini
plugins: timeout-2.2.0, anyio-4.3.0
collected 181 items                                                                                                                   

kadi/commands/tests/test_commands.py .................................................................................          [ 44%]
kadi/commands/tests/test_states.py .......................x.........................                                            [ 71%]
kadi/commands/tests/test_validate.py ...................                                                                        [ 82%]
kadi/tests/test_events.py ..........                                                                                            [ 87%]
kadi/tests/test_occweb.py ......................                                                                                [100%]

============================================= 180 passed, 1 xfailed in 218.51s (0:03:38) ==============================================
(ska3-masters) jeanconn-fido> git rev-parse HEAD
587015c61df212f2073c8d5e827181265b99f447
```
### Functional tests

Functional testing is available in the included validation Jupyter notebook. That code is mostly superseded by the unit test in #340, however it is worth having since it shows a bit more detail.

I also confirmed that the validation plots look as expected. This is not a surprise since they just reflect the states.

### CODE REVIEW

File: kadi/commands/states.py [hunk 1]
```diff
@@ -1387,24 +1387,43 @@ class NormalSunTransition(ManeuverTransition):
...
         """
         This is a transition function callback.
 
         It directly sets the state pcad_mode to NSUN and target quaternion to
         the expected sun pointed attitude.  It then calls the parent method to
         add the actual maneuver.
         """
-        # Transition to NSUN
-        state["pcad_mode"] = "NSUN"
+        # Transition to approprpriate PCAD mode
```
Comment (jeanconn) [inline kadi/commands/states.py:1425]: Typo.

**Comment by jeanconn** [comment]
For me this isn't passing unit tests on linux or mac.

**Comment by taldcroft** [comment]
@jeanconn - seems this change is pretty impacting. Most tests are passing now but there are still issues with the validation regression. I need to step away to a different task now so this needs to wait for the moment.

**Comment by jeanconn** [comment]
For validation, I note that I couldn't get tests passing when gave a 3 minute shot a regenerating validation regress data in #338 .  I bring that up in that, heck, we might want to add that in as *another* change in this PR to only update the regress data once...

**Comment by taldcroft** [comment]
Some validate regression tests are mostly failing, but with occasionally successes. Digging required...

File: kadi/commands/tests/data/validator_pcad_mode_2022297_5_False.yaml [hunk 2]
```diff
@@ -72,6 +72,8 @@ vals_compressed:
   - 1.0
   - 3.0
   - 3.0
+  - 0.0
+  - 0.0
```
Comment (taldcroft) [inline kadi/commands/tests/data/validator_pcad_mode_2022297_5_False.yaml:76]: This is pcad_mode=STBY, aka Safe Mode

File: kadi/commands/tests/test_commands.py [hunk 3]
```diff
@@ -624,7 +624,6 @@ def test_command_set_safe_mode():
     exp = """\
 2000:001:00:00:00.000 | COMMAND_SW       | ACPCSFSU   | CMD_EVT  | event=Safe_mode, event_date=2000:001:00:00:00, scs=0
 2000:001:00:00:00.000 | COMMAND_SW       | CSELFMT5   | CMD_EVT  | event=Safe_mode, event_date=2000:001:00:00:00, scs=0
-2000:001:00:00:00.000 | COMMAND_SW       | AONSMSAF   | CMD_EVT  | event=Safe_mode, event_date=2000:001:00:00:00, scs=0
```
Comment (taldcroft) [inline kadi/commands/tests/test_commands.py:627]: Command events no longer inserts a bogus NSM transition.

Comment (jeanconn) [reply kadi/commands/tests/test_commands.py:627]: That seems not a "support offset pitch" change?

Comment (taldcroft) [reply kadi/commands/tests/test_commands.py:627]: I updated the PR title and description to reflect the broader scope of this PR. Good catch.

File: kadi/commands/states.py [hunk 4]
```diff
@@ -1388,24 +1387,43 @@ class NormalSunTransition(ManeuverTransition):
...
+            transitions[cmd["date"]]["maneuver_transition"] = functools.partial(
+                cls.callback, cls, pitch
+            )
+
+    @staticmethod
+    def callback(cls, pitch, date, transitions, state, idx):  # noqa: PLW0211
         """
         This is a transition function callback.
 
         It directly sets the state pcad_mode to NSUN and target quaternion to
```
Comment (jeanconn) [inline kadi/commands/states.py:1421]: Does this docstring also need a tweak for appropriate PCAD mode instead of NSUN?

Comment (taldcroft) [reply kadi/commands/states.py:1421]: Done.

**Review by jeanconn — APPROVED** [review]

**Pull request merged**